django_forest/authentication/oidc/client_manager.py

```diff
@@ -0,0 +1,46 @@
+from urllib.parse import urljoin
+
+from oic.oic import Client
+from oic.oic.message import ProviderConfigurationResponse
+
+from .configuration_retriever import retrieve
+from .dynamic_client_registrator import register
+
+
+class OidcClientManager:
+    client = None
+
+    @classmethod
+    def get_client_for_callback_url(cls, callback_url):
```

Comment:
`get_client_for_callback_url` made me think that `OidcClientManager` would hold multiple instances to handle multiple callback URLs.

Reply:
I've seen that `forest-express` and `forest-rails` use a cache, but with a size of 1... Always on the same key.
So I decided yo use a singleton here.